stl/inc/syncstream

```diff
@@ -0,0 +1,339 @@
+        _STD swap(_Wrapped, _Right._Wrapped);
+    }
+
+    bool _Emit() noexcept {
+        _TRY_BEGIN
+        return emit();
+        _CATCH_ALL
+        return false;
+        _CATCH_END
+    }
+
+    _NODISCARD constexpr size_type _Calculate_growth(
+        const size_type _Oldsize, const size_type _Newsize, const size_type _Maxsize) const {
+        if (_Oldsize > _Maxsize - _Oldsize / 2) {
+            return _Maxsize; // geometric growth would overflow
+        }
+
+        const size_type _Geometric = _Oldsize + _Oldsize / 2;
+
+        if (_Geometric < _Newsize) {
+            return _Newsize; // geometric growth would be insufficient
+        }
+
+        return _Geometric; // geometric growth is sufficient
+    }
+
+    _NODISCARD constexpr size_type _Get_data_size() const noexcept {
+        return static_cast<size_type>(streambuf_type::pptr() - streambuf_type::pbase());
+    }
+
+    _NODISCARD constexpr size_type _Get_buffer_size() const noexcept {
+        return static_cast<size_type>(streambuf_type::epptr() - streambuf_type::pbase());
+    }
+
+    _NODISCARD constexpr _Alloc& _Getal() noexcept {
+        return _Al;
+    }
+
+    _NODISCARD constexpr const _Alloc& _Getal() const noexcept {
+        return _Al;
+    }
+
+    streambuf_type* _Wrapped{nullptr};
+    allocator_type _Al{};
+    bool _Emit_on_sync{false};
+    bool _Record_sync{false};
+};
+
+template <class _Elem, class _Traits, class _Alloc>
+void swap(basic_syncbuf<_Elem, _Traits, _Alloc>& _Left, basic_syncbuf<_Elem, _Traits, _Alloc>& _Right) {
+    _Left.swap(_Right);
+}
+
+template <class _Elem, class _Traits, class _Alloc>
+class basic_osyncstream : public basic_ostream<_Elem, _Traits> {
+public:
+    using char_type      = _Elem;
+    using int_type       = typename _Traits::int_type;
+    using pos_type       = typename _Traits::pos_type;
+    using off_type       = typename _Traits::off_type;
+    using traits_type    = _Traits;
+    using allocator_type = _Alloc;
+    using streambuf_type = basic_streambuf<_Elem, _Traits>;
+    using syncbuf_type   = basic_syncbuf<_Elem, _Traits, _Alloc>;
+
+    using _Mybase = basic_ostream<_Elem, _Traits>;
+
+    basic_osyncstream(streambuf_type* _Strbuf, const _Alloc& _Al)
+        : _Mybase(_STD addressof(_Sync_buf)), _Sync_buf(_Strbuf, _Al) {}
+
+    explicit basic_osyncstream(streambuf_type* _Strbuf) : basic_osyncstream(_Strbuf, _Alloc{}) {}
+
+    basic_osyncstream(basic_ostream<_Elem, _Traits>& _Str, const _Alloc& _Al) : basic_osyncstream(_Str.rdbuf(), _Al) {}
+
+    explicit basic_osyncstream(basic_ostream<_Elem, _Traits>& _Str) : basic_osyncstream(_Str, _Alloc{}) {}
+
+    basic_osyncstream(basic_osyncstream&& _Right) noexcept
+        : _Mybase(_STD move(_Right)), _Sync_buf(_STD move(_Right._Sync_buf)) {
+        _Mybase::set_rdbuf(_STD addressof(_Sync_buf));
+    }
+
+    ~basic_osyncstream() {
+        _Emit();
+    }
+
+    basic_osyncstream& operator=(basic_osyncstream&&) noexcept = default;

```

Comment:
[LWG-3334](https://cplusplus.github.io/LWG/issue3334) removed the assignment operator's `Effect` and `Postconditions`, I am wondering if this means that the function should be `default` or be removed since it exists in the current working draft's header overview only.

Reply:
WG21-N4878 16.3.3.4 [functions.within.classes]/1 "For the sake of exposition, Clause 17 through Clause 32 and Annex D do not describe copy/move constructors, assignment operators, or (non-virtual) destructors with the same apparent semantics as those that can be generated by default (11.4.5.3, 11.4.6, 11.4.7). It is unspecified whether the implementation provides explicit definitions for such member function signatures, or for virtual destructors that can be generated by default."

Defining it with `= default` seems reasonable.